Update to egui 0.17

bobtwinkles opened the pull request:
I'm not entirely sure that the sampler injection (`egui_texture_from_wgpu_texture` and friends) works, I haven't tested it yet.

Fixes #56

asayers commented:
I tested this branch; it seems to work.  (Didn't test user textures.)

hasenbanck commented:
Thank you for the PR. Some comments I have:

1. You pulled in the log crate as dependency for something that seems like debug output. egui itself also optionaly uses the tracing crate. Please remove the dependency or make it also optional based on tracing so that it aligned with the upstream project.
2. Please don't use unwrap(). If you think it can fail, than please return an error. You also seem to test if the texture sizes are bigger than an u32, which is fine, but then it should return a custom error instead of panicing. I also don't think that this error can be practically be thrown, since WGPU will most likely throw a "Error::Validation" before, since device limits are much lower anyhow.

bobtwinkles commented:
> You pulled in the log crate as dependency for something that seems like debug output.

Removed. I added it this way to align with `wgpu` conventions (which exposes detailed information via `log`).

>  Please don't use unwrap().

I don't expect most of these to get hit in practice either, so I've ripped them out.

hasenbanck commented on src/lib.rs:
```diff
@@ -360,131 +362,143 @@ impl RenderPass {
+                egui::ImageData::Alpha(_) => {
+                    // The unwrap here should never fail as alpha_srgb_pixels will have been set to
+                    // `Some` above.
+                    bytemuck::cast_slice(alpha_srgb_pixels.as_ref().unwrap().as_slice())
```
Even if it's impossible in the current state of the code, don't throw a generic panic and at least specify the panic reason  with "expect()".

hasenbanck commented:
Looks fine now. I will give people time to test the changes and then release a now version to cargo in some days.

bobtwinkles commented:
Thanks for the review here and on the `winit` integration, take care!
